toolchain/lower/handle.cpp

```diff
@@ -168,10 +167,38 @@ auto HandleBuiltin(FunctionContext& /*context*/, SemIR::InstId /*inst_id*/,
   CARBON_FATAL() << "TODO: Add support: " << inst;
 }
 
+// Get the predicate to use for an `icmp` instruction generated for the
+// specified builtin.
+static auto GetBuiltinICmpPredicate(SemIR::BuiltinFunctionKind builtin_kind,
+                                    bool is_signed)
+    -> llvm::CmpInst::Predicate {
+  switch (builtin_kind) {
+    case SemIR::BuiltinFunctionKind::IntEq:
+      return llvm::CmpInst::ICMP_EQ;
+    case SemIR::BuiltinFunctionKind::IntNeq:
+      return llvm::CmpInst::ICMP_NE;
+    case SemIR::BuiltinFunctionKind::IntLess:
+      return is_signed ? llvm::CmpInst::ICMP_SLT : llvm::CmpInst::ICMP_ULT;
+    case SemIR::BuiltinFunctionKind::IntLessEq:
+      return is_signed ? llvm::CmpInst::ICMP_SLE : llvm::CmpInst::ICMP_ULE;
+    case SemIR::BuiltinFunctionKind::IntGreater:
+      return is_signed ? llvm::CmpInst::ICMP_SGT : llvm::CmpInst::ICMP_UGT;
+    case SemIR::BuiltinFunctionKind::IntGreaterEq:
+      return is_signed ? llvm::CmpInst::ICMP_SGE : llvm::CmpInst::ICMP_UGE;
+    default:
+      CARBON_FATAL() << "Unexpected builtin kind " << builtin_kind;
+  }
+}
+
 // Handles a call to a builtin function.
 static auto HandleBuiltinCall(FunctionContext& context, SemIR::InstId inst_id,
                               SemIR::BuiltinFunctionKind builtin_kind,
                               llvm::ArrayRef<SemIR::InstId> arg_ids) -> void {
+  auto is_signed = [&](SemIR::InstId int_id) {
+    return context.sem_ir().types().IsSignedInt(
+        context.sem_ir().insts().Get(int_id).type_id());
+  };
```

Comment:
FWIW this feels like it could be a small helper function instead, `static auto IsSignedInt(FunctionContext& context, SemIR::InstId inst_id)`

Reply:
Done.